### DISCUSSION TITLE
Using Sumatra as the default PDF Viewer in IE11 and or Edge

### DISCUSSION THREAD

**Discussion by nmulconray** [sumatrapdfreader/sumatrapdf#2404] [Q&A]
Hi,
I use creative cloud Adobe DC and the reader however I have to wait up to ten minutes sometimes to view a pdf and now its updating my pdf reader 2018 without my permission, so I cant instantly open PDFs anymore(the logon process is tedious and time consuming). This changes the default and ruins my ability to open PDFs.
I was wondering if I could set Sumatra as the default in IE11 as it's light and fast!
cheers
Nik

**Comment by GitHubRulesOK** [comment]
SumatraPDF only worked with Firefox derivatives and currently only fast 32bit browser I can use is Palemoon and for 64bit I can also use Waterfox classic. So short answer is generally the legacy support for SumatraPDF plugin is very limited to those 3.

**Comment by Ander3232** [comment]
Why oh why doesn't Sumatra, like virtually every other modern app that opens particular file types, include a setting to make it the default PDF reader? I've found _no way_ to make such a setting in Windows. From Windows' viewpoint, it's as though Sumatra doesn't exist. Microsoft is apparently intent on keeping PDFs opening in MS Edge. It's ridiculous.

**Reply by GitHubRulesOK** [reply]
@Ander3232 

Correct Firefox, Foxit, Others and even Adobe have had to devote teams to overcome the way MS keep moving UAC settings, the silly point is the encryption for User preferences is fairly simple to circumvent so MS retrigger(s) Edge as dominated PDF default. A clean reset Adobe is about 150 registry settings ! but primary entry can be seen with   
 `REG QUERY  HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.pdf\UserChoice` that should show MSEDGE as UAC application

The best way to reset is USER select ANY PDF press ALT/Enter OR right click PROPERTIES then change the "open with" then ignore Edge and other old assignments **including ignore any SumatraPDF icons** as potential old entries keep move down to select **always open AND LOOK for another app on PC** then pick the EXE you want e.g. either  in %localappdata% or %programfiles% sumatrapdf e.g. which ever has the **desired sumatrapdf.exe** now if you check you should see similar to

```
> REG QUERY HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.pdf\UserChoice

HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.pdf\UserChoice
    ProgId    REG_SZ    SumatraPDF.pdf
    Hash    REG_SZ    ugfTZgIpvT8=
```

the problem is the Hash can change each time so as example resetting  

    ProgId    REG_SZ    MSEdgePDF
    Hash    REG_SZ    **2YAsaliqJuU=**

    ProgId    REG_SZ    SumatraPDF.pdf
    Hash    REG_SZ    ugfTZgIpvT8=

    ProgId    REG_SZ    MSEdgePDF
    Hash    REG_SZ    **Tn9SNtvdjd4=**

thus varies each time

there is an exe and powershell script that aims to reset by hashing and it is interesting that the SFTA tool uses SumatraPDF as an example (***this is NOT a proposal or endorsement*** just illustration of typical use)

*Register Application + Set File Type Association*  
   SFTA.exe --reg "C:\SumatraPDF.exe" ".PDF"  
   SFTA.exe --reg "C:\SumatraPDF.exe" ".PDF" "CustomProgramId"  
  
   *Register Application + Set Protocol Association*  
   SFTA.exe --reg "C:\SumatraPDF.exe" "http"  
  
   *Unregister Application*  
   SFTA.exe --unreg "C:\SumatraPDF.exe" ".PDF"  
   SFTA.exe --unreg "CustomProgramId" ".PDF"*

Simple command was ***SFTA.exe Applications\SumatraPDF.exe .pdf*** but you need to experiment.  
Most recent (22 days ago) PowerShell version https://github.com/DanysysTeam/PS-SFTA  
older exe https://danysys.com/set-file-type-association-default-application-command-line-windows-10-userchoice-hash-internal-method/

**Reply by Ander3232** [reply]
> The best way to reset is USER select ANY PDF press ALT/Enter OR right click PROPERTIES then change the "open with" then ignore Edge and other old assignments **including ignore any SumatraPDF icons** as potential old entries keep move down to select **always open AND LOOK for another app on PC** then pick the EXE you want...

Yup, tried that, of course. Several times. Result: _That_ particular PDF opened in Sumatra—but the default app (Edge) remained unchanged, even though I selected "Always use this app to open PDF files".

The good news is, I think I've solved the problem (until MS gets busy again, LOL)! I switched from the 64-bit Sumatra to the 32-bit, and Windows has accepted it as the default PDF app. The 64-bit world has always seemed to have some lingering flakiness; I've solved various problems by backpedaling on various other 64-bit apps (on various PCs). Whatever works though, right? 🙋🏽‍♂️u

**Comment by GitHubRulesOK** [comment]
2026 using SumatraPDF in a browser the choice is now really limited to 2 :-)

so for the last man standing https://www.palemoon.org/ and it's capable spin off  https://www.basilisk-browser.org 
<img width="663" height="224" alt="image" src="https://github.com/user-attachments/assets/b70121b3-0c5e-4c55-a890-f8860bed4388" />

 see 
https://github.com/GitHubRulesOK/SumatraPDF-Plus/tree/master/2.5%20Legacy%20Browser%20Plugin

<img width="1228" height="604" alt="Pale Moon" src="https://github.com/user-attachments/assets/830b1a57-6659-4d9b-be27-1f4872967b72" />
